### PR TITLE
over window function

### PR DESCRIPTION
**Pull request by iamlinjunhong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #8805

## What this PR does / why we need it:

doc: https://github.com/matrixorigin/docs/blob/main/design/window%20func.md

support window func: avg, count, max, sum, min, rank

### CODE REVIEW

**Comment by iamlinjunhong** [comment]
> [n/window%20func.md](https://github.com/matrixorigin/docs/blob/main/design/window%20func.md)
> 
> support window func: avg, count, max, sum, min, rank

I didn't see row_ number in tpc-ds, but row_ number is quite easy to implement, it can be implemented before 0.8

**Comment by iamlinjunhong** [comment]
> Only rank? Row_number at least?

I didn't see row_ number in tpc-ds, but row_ number is quite easy to implement, it can be implemented before 0.8

**Review by XuPeng-SH — APPROVED** [review]

File: pkg/container/types/types.go [hunk 1]
```diff
@@ -461,6 +461,27 @@ func (t Type) IsFloat() bool {
 	}
 }
 
+func (t Type) IsDecimal() bool {
```
[the remark this answers is not in the document]
Comment (iamlinjunhong) [reply pkg/container/types/types.go:464]: 统一用 type 判断了，Oid 判断和 Type 判断冗余了，应该全部统一。

File: pkg/sql/colexec/window/window.go [hunk 2]
```diff
@@ -0,0 +1,987 @@
...
+	ap := arg.(*Argument)
+	ctr := ap.ctr
+	anal := proc.GetAnalyze(idx)
+	anal.Start()
+	defer anal.Stop()
+
+	for {
+		bat, end, err := ctr.ReceiveFromAllRegs(anal)
+		if err != nil || end {
+			break
```
Comment (m-schen) [inline pkg/sql/colexec/window/window.go:70]: if err != nil {
  return false, err
}
if end {
  break
}

Comment (iamlinjunhong) [reply pkg/sql/colexec/window/window.go:73]: fixed

File: pkg/sql/plan/base_binder.go [hunk 3]
```diff
@@ -883,9 +883,9 @@ func (b *baseBinder) bindFuncExpr(astExpr *tree.FuncExpr, depth int32, isRoot bo
 	}
 	funcName := funcRef.Parts[0]
 
-	if function.GetFunctionIsAggregateByName(funcName) {
+	if function.GetFunctionIsAggregateByName(funcName) && astExpr.WindowSpec == nil {
 		return b.impl.BindAggFunc(funcName, astExpr, depth, isRoot)
-	} else if function.GetFunctionIsWinFunByName(funcName) {
+	} else if function.GetFunctionIsWinFunByName(funcName) && astExpr.WindowSpec != nil {
```
Comment (aunjgr) [inline pkg/sql/plan/base_binder.go:888]: Should this `&&` be `||`?

**Review by aressu1985 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]
plan part and result looks good to me

**Review by m-schen — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Pull request merged**